test/installer/scripts/prep_omsadmin.sh

```diff
@@ -22,6 +22,9 @@ sed -i s,AGENT_GROUP=.*,AGENT_GROUP=`id -gn`,1 $OMSADMIN
 sed -i s,METACONFIG_PY=.*,METACONFIG_PY=$TESTDIR/not_a_file,1 $OMSADMIN
 sed -i s,PROCESS_STATS=.*,PROCESS_STATS=$TESTDIR/process_stats,1 $OMSADMIN
 
+# Don't run service_control in a test
+sed -i s,/opt/microsoft/omsagent/bin/service_control[.]*,,1 $OMSADMIN
```

Comment:
Oh, I don't like this one bit, and it's a bad idea:

- It's confusing that we change a shipping file from what is checked out,
- Nothing changes it back (that I see), possibly causing staging problems and shipping the wrong version

If service_control is being run as part of a test, then I think running a test should set an environment variable and whoever is calling service_control shouldn't do that if the "OMSAGENT_TESTING_IN_PROGRESS" is set. Or something like that.